src/calendar/CalendarView.js

```diff
@@ -272,11 +276,96 @@ export class CalendarView implements CurrentView {
 			    .then(() => this._loadMonthIfNeeded(previousMonthDate))
 		})
 
+		this._setupShortcuts();
+
 		locator.eventController.addEntityListener((updates, eventOwnerGroupId) => {
 			this.entityEventReceived(updates, eventOwnerGroupId)
 		})
 	}
 
+	_setupShortcuts() {
+		let shortcuts: Shortcut[] = [
+			{
+				key: Keys.ONE,
+				exec: () => this._setUrl(CalendarViewType.DAY, this.selectedDate()),
+				help: "switchDayView_action"
+			},
+			{
+				key: Keys.TWO,
+				exec: () => this._setUrl(CalendarViewType.WEEK, this.selectedDate()),
+				help: "switchWeekView_action"
+			},
+			{
+				key: Keys.THREE,
+				exec: () => this._setUrl(CalendarViewType.MONTH, this.selectedDate()),
+				help: "switchMonthView_action"
+			},
+			{
+				key: Keys.FOUR,
+				exec: () => this._setUrl(CalendarViewType.AGENDA, this.selectedDate()),
+				help: "switchAgendaView_action"
+			},
+			{
+				key: Keys.T,
+				exec: () => this._setUrl(m.route.param("view"), new Date()),
+				help: "viewToday_action"
+			},
+			{
+				key: Keys.N,
+				enabled: () => this._currentViewType !== CalendarViewType.AGENDA,
+				exec: () => this._viewPeriod(true),
+				help: "viewNextPeriod_action"
+			},
+			{
+				key: Keys.P,
+				enabled: () => this._currentViewType !== CalendarViewType.AGENDA,
+				exec: () => this._viewPeriod(false),
+				help: "viewPrevPeriod_action"
+			},
+			{
+				key: Keys.C,
+				ctrl: true,
+				exec: () => this._showCreateCalendarDialog(),
+				help: "createCalendar_action"
+			},
+			{
+				key: Keys.F,
+				exec: () => this._newEvent(this.selectedDate()),
+				help: "createEvent_action"
+			}
+		]
+
+		this.oncreate = () => keyManager.registerShortcuts(shortcuts)
+		this.onbeforeremove = () => keyManager.unregisterShortcuts(shortcuts)
+	}
+
+	_viewPeriod(next) {
+		let newDate = new Date(this.selectedDate().getTime())
```

Comment:
This duplicates code from `onChangeWeek` and the like, I think we should extract it

Reply:
I used the `_viewPeriod()` in `onChangeWeek` as well as `onChangeMonth`